gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/InitialImageOperation.java

```diff
@@ -416,6 +442,11 @@ GIIStatus getFromOne(
         int processorType = targetReinitialized ? DistributionManager.WAITING_POOL_EXECUTOR
             : DistributionManager.HIGH_PRIORITY_EXECUTOR;
         try {
+
+          if (internalNoGII != null && internalNoGII.getRegionName().equals(region.getName())) {
```

Comment:
If we the test hook we are interrupting, is it intended that the test at least goes to the next one and does gii? If yes then won't the gii will fail if we issue interrupt to the current thread.